File: wrappers/kotlin/src/commonTest/kotlin/CryptoBoxTest.kt
package tech.indicio.holdr

import askar.crypto.CryptoBox
import askar.crypto.Key
import askar.enums.KeyAlgs
import kotlinx.cinterop.toKString
import kotlin.test.Test
import kotlin.test.assertEquals

class CryptoBoxTest {

    @Test
    fun seal() {
        val x25519Key = Key.generate(KeyAlgs.X25519)

        val message = "foobar"
        val sealed = CryptoBox.seal(x25519Key, message)

        val opened = CryptoBox.sealOpen(x25519Key, sealed)
        assertEquals(message, opened.toKString())

        x25519Key.handle().free()
    }
}

File: wrappers/javascript/packages/aries-askar-nodejs/tests/cryptoBox.test.ts
import { CryptoBox, Key, KeyAlgs } from '@hyperledger/aries-askar-shared'

import { setup } from './utils/initialize'

describe('CryptoBox', () => {
  beforeAll(setup)

  test('seal', () => {
    const x25519Key = Key.generate(KeyAlgs.X25519)

    const message = Uint8Array.from(Buffer.from('foobar'))
    const sealed = CryptoBox.seal({ recipientKey: x25519Key, message })

    const opened = CryptoBox.sealOpen({
      recipientKey: x25519Key,
      ciphertext: sealed,
    })
    expect(opened).toStrictEqual(message)
    x25519Key.handle.free()
  })
})
